File: Java/Medium/WrongAnswer/concatenation-of-consecutive-binary-numbers_9.java
class Solution {
    public int concatenatedBinary(int n) {
        
        int MOD = 1000000007;
        int result = 0;
        for (int i = 1; i <= n; i++) {
            int len = 0;
            int temp = i;
            while (temp > 0) {
                temp >>= 1;
                len++;
            }
            result = ((result << len) | i) % MOD;
        }
        return result;
    }
}

File: Rust/Medium/WrongAnswer/concatenation-of-consecutive-binary-numbers_9.rs
impl Solution {
    pub fn concatenated_binary(n: i32) -> i32 {
        
        let mut result = 0;
        let mut mod_ = 1_000_000_007;
        for i in 1..=n {
            let mut x = i;
            let mut num = 0;
            while x > 0 {
                x >>= 1;
                num += 1;
            }
            result = (result << num) % mod_ + i as i32;
            result %= mod_;
        }
        result
    }
}